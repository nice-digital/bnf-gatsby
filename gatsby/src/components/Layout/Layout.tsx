import React, { ReactNode } from "react";

import { Footer, Main } from "@nice-digital/global-nav";
import { Container } from "@nice-digital/nds-container";

import { useIsClient } from "@/hooks/useIsClient";

import { IEBanner } from "../IEBanner/IEBanner";
import { SiteDistinction } from "../SiteDistinction/SiteDistinction";
import { SiteHeader } from "../SiteHeader/SiteHeader";

import { isBNF } from "./../../site";
import "./../../styles/feed.scss";
import "./../../styles/global.scss";

type LayoutProps = {
	children: ReactNode;
};

declare global {
	interface Document {
		documentMode?: unknown;
	}
}

export const Layout: React.FC<LayoutProps> = ({ children }: LayoutProps) => {
	const isClient = useIsClient();
	let isIE = false;
	if (isClient) {
		const doc: Document = window.document;
		isIE = !!doc.documentMode;
	}

	return (
		<>
			{isIE && <IEBanner />}
			<SiteHeader />
			<Main>
				<SiteDistinction />
				<Container>{children}</Container>
				<div id="sticky-nav-portal"></div>
			</Main>
			<Footer service={isBNF ? "bnf" : "bnfc"} />
		</>
	);
};
