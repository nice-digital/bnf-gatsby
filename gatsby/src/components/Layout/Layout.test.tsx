import { render, screen } from "@testing-library/react";

import { Layout } from "@/components/Layout/Layout";

describe("Layout", () => {
	it("should match snapshot", () => {
		const { container } = render(
			<Layout>
				<div>Content</div>
			</Layout>
		);
		expect(container).toMatchSnapshot();
	});

	it("Should not render the browser banner by default", () => {
		render(
			<Layout>
				<div>Content</div>
			</Layout>
		);

		expect(
			// eslint-disable-next-line testing-library/prefer-presence-queries
			screen.queryByRole("heading", { level: 2, name: "Browser support" })
		).toBeNull();
	});

	it("Should render a banner whenever IE11 is detected", () => {
		Object.defineProperty(global.document, "documentMode", { value: "test" }); // Spoof IE by adding a documentMode property to the document object

		render(
			<Layout>
				<div>Content</div>
			</Layout>
		);

		expect(
			screen.getByRole("heading", { level: 2, name: "Browser support" })
		).toBeInTheDocument();
	});
});
