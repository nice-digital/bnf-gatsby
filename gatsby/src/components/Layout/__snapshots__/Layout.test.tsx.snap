// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should match snapshot 1`] = `
<div>
  <main>
    <div
      class="container"
      data-component="container"
    >
      <div>
        Content
      </div>
    </div>
    <div
      class="panel"
    >
      <div
        class="container"
      >
        <div
          class="panel"
          data-component="panel"
        >
          <p>
            The content on the NICE BNFC site (BNFC) is the copyright of BMJ Publishing Group Ltd, the Royal Pharmaceutical Society of Great Britain, and RCPCH Publications Ltd. By using BNFC, you agree to the licence set out in the
             
            <a
              href="/eula"
            >
              BNF Publications End User Licence Agreement
            </a>
            .
          </p>
        </div>
      </div>
    </div>
    <div
      id="sticky-nav-portal"
    />
  </main>
</div>
`;
